### PR TITLE
Remove hifi screenshare (updated) (updated)

### PR DESCRIPTION
**Pull request by HifiExperiments**
PR #757, rebased on the protocol_changes branch

since adding/removing properties is now much simpler, I took the opportunity to rebase this PR once again and truly remove the property and bump the protocol

### CODE REVIEW

File: libraries/networking/src/udt/PacketHeaders.h [hunk 1]
```diff
@@ -341,7 +341,8 @@ enum class EntityVersion : PacketVersion {
     ShadowBiasAndDistance,
     TextEntityFonts,
     ScriptServerKinematicMotion,
-    ScreenshareZone,
+    // This was ScreenshareZone, property was dropped.
+    ScreenshareZoneUnused,
```
Comment (ksuprynowicz) [inline libraries/networking/src/udt/PacketHeaders.h:345]: Do we need this if we are doing a protocol change?

File: libraries/networking/src/udt/PacketHeaders.h [hunk 2]
```diff
@@ -362,6 +363,7 @@ enum class EntityVersion : PacketVersion {
     ModelLoadPriority,
     PropertyCleanup,
     TextVerticalAlignment,
+    RemoveScreenshare,
```
Comment (ksuprynowicz) [inline libraries/networking/src/udt/PacketHeaders.h:366]: Same as above

File: libraries/networking/src/udt/PacketHeaders.h (hunk 1 above)
Comment (HifiExperiments) [reply libraries/networking/src/udt/PacketHeaders.h:345]: if we remove this, it decrements all of the values after, which means when the next protocol change is made, the last value will be the same as it is now, but with an incompatible protocol, which will cause crashes.  renaming this isn’t really important, it’s just left over from the previous commits, but it doesn’t hurt

File: libraries/networking/src/udt/PacketHeaders.h (hunk 2 above)
Comment (HifiExperiments) [reply libraries/networking/src/udt/PacketHeaders.h:366]: when properties are removed, we still need to bump the protocol to make sure incompatible versions can’t connect to each other.  even if we’re combining this with other protocol bumps, it’s still safer to just bump it again

Comment (ksuprynowicz) [reply libraries/networking/src/udt/PacketHeaders.h:366]: Thanks for explaining these!

**Review by ksuprynowicz — APPROVED** [review]
Everything looks fine :)

**Comment by ksuprynowicz** [comment]
I tested it and everything works fine :)

**Pull request merged**